src/main/features/directions-questionnaire/forms/models/availability.ts

```diff
@@ -0,0 +1,58 @@
+import { ArrayNotEmpty, IsDefined, ValidateIf, ValidateNested } from '@hmcts/class-validator'
+import { ValidationErrors as GlobalValidationErrors } from 'forms/validation/validationErrors'
+import { LocalDate } from 'forms/models/localDate'
+import { IsValidLocalDate } from '@hmcts/cmc-validators'
+import { IsFutureDate } from 'forms/validation/validators/dateFutureConstraint'
+
+export class ValidationErrors {
+  static readonly AT_LEAST_ONE_DATE: string = 'Select at least one date or choose No'
+  static readonly DATE_NOT_VALID: string = 'Please enter a valid date'
+}
+
+export class Availability {
+
+  @IsDefined({ message: GlobalValidationErrors.YES_NO_REQUIRED })
+  hasUnavailableDates?: boolean
+
+  @ValidateIf(o => o.hasUnavailableDates)
+  @ValidateNested()
+  @ArrayNotEmpty({ message: ValidationErrors.AT_LEAST_ONE_DATE })
+  @IsValidLocalDate({ message: ValidationErrors.DATE_NOT_VALID, each: true })
+  unavailableDates?: LocalDate[]
```

Comment:
Requirement from Marc to validate the array _is_ empty when the user says they don't want to declare any.